pkg/action/cobra.go

```diff
@@ -23,28 +23,35 @@ func CobraImpl(a *Action, streams cli.Streams) (*cobra.Command, error) {
 	options := make(TypeOpts)
 	runOpts := make(TypeOpts)
 	cmd := &cobra.Command{
-		Use:  use,
-		Args: cobra.ExactArgs(len(argsDef)),
+		Use: use,
+		// Using custom args validation in ValidateInput.
 		// @todo: maybe we need a long template for arguments description
 		Short: getDesc(actConf.Title, actConf.Description),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cmd.SilenceUsage = true // Don't show usage help on a runtime error.
 			// Pass to the run environment its flags.
 			if env, ok := a.env.(RunEnvironmentFlags); ok {
+				runOpts = filterFlags(cmd, runOpts)
 				err := env.UseFlags(derefOpts(runOpts))
 				if err != nil {
 					return err
 				}
 			}
 			// Set action input.
 			err := a.SetInput(Input{
-				Args: argsToMap(args, argsDef),
-				Opts: derefOpts(options),
-				IO:   streams,
+				Args:    argsToMap(args, argsDef),
+				Opts:    derefOpts(options),
+				IO:      streams,
+				ArgsRaw: args,
 			})
 			if err != nil {
 				return err
 			}
+
+			if err = a.env.(RunEnvironmentFlags).ValidateInput(a); err != nil {
```

Comment:
It's not correct to run it like that.
First, you have to type assert then run the Validate or it will fail.
Validate must be run before SetInput.

Reply:
Please, test the case when command is defined like that:
```
command:
  - {{ .myArg1 }}
```

And action is run `launchr my.action --exec -- ls -al`

Reply:
not implemented ?